unit_testing/test_register.py

```diff
@@ -167,11 +189,36 @@ def test_register_step_ants_registration(step2_data):
      verbose=cli_params.verbose,
     )
 
-@pytest.mark.skip(reason="TODO")
-def test_register_step_ants_slice_regularized_registration():
-     """
-     """
-     raise NotImplementedError()
+
+def test_register_step_ants_slice_regularized_registration(step_axial_data_in_same_space):
+    """
+    Read CSV file, remove first element, flatten and convert to float
+    """
+    def csv2num(file_csv):
+        with open(file_csv, newline='') as csvfile:
+            spamreader = list(csv.reader(csvfile, delimiter=','))
+        # Remove first element corresponding to header ['Tx', 'Ty']
+        spamreader.pop(0)
+        # Flatten
+        spamreader = [item for sublist in spamreader for item in sublist]
+        # Convert each element to float
+        spamreader = [float(i) for i in spamreader]
+        return spamreader
```

Comment:
So I poked around and it turns out this is built into numpy as `np.genfromtxt`. I'm a little aghast I didn't know it before:

```
$ cat /tmp/a.csv 
Tx,Ty
1.00439255454324,-0.151760183772956
1.04645709177172,-0.621178997815087
0.904949339396005,-0.862157494292993
0.784728781515325,-0.893095039021259
0.890654902228898,-0.732390997814474
$ python
>>> numpy.genfromtxt("/tmp/a.csv", names=True, delimiter=",")
array([(1.00439255, -0.15176018), (1.04645709, -0.621179  ),
       (0.90494934, -0.86215749), (0.78472878, -0.89309504),
       (0.8906549 , -0.732391  )], dtype=[('Tx', '<f8'), ('Ty', '<f8')])
>>> A = _
>>> A['Tx']
array([1.00439255, 1.04645709, 0.90494934, 0.78472878, 0.8906549 ])
>>> A[0]
(1.00439255, -0.15176018)
>>> 
>>> 
>>> numpy.genfromtxt("/tmp/a.csv", skip_header=1, delimiter=",")
array([[ 1.00439255, -0.15176018],
       [ 1.04645709, -0.621179  ],
       [ 0.90494934, -0.86215749],
       [ 0.78472878, -0.89309504],
       [ 0.8906549 , -0.732391  ]])
>>> A = _
>>> A[0]
array([ 1.00439255, -0.15176018])
>>> A[0,1]
-0.151760183772956
```

Notice how it understands CSV headers, if you ask it to.

There's also `np.fromfile` but it seems a bit pickier, and `np.loadtxt`, which don't understand headers:

```
>>> numpy.loadtxt("/tmp/a.csv", skiprows=1, delimiter=",")
array([[ 1.00439255, -0.15176018],
       [ 1.04645709, -0.621179  ],
       [ 0.90494934, -0.86215749],
       [ 0.78472878, -0.89309504],
       [ 0.8906549 , -0.732391  ]])
```

Reply:
AMAZING! i'll do that--- thanks!